crates/sui-storage/src/indexes.rs

```diff
@@ -885,18 +886,49 @@ impl IndexStore {
         name: &DynamicFieldName,
```

Comment:
Pre-existing, but would you mind changing this to `DynamicFieldName`? All the call sites have ownership, and it will save us a type clone